controllers/topology/reconcile_state.go

```diff
@@ -397,8 +398,8 @@ func (r *ClusterReconciler) reconcileReferencedObject(ctx context.Context, curre
 	}
 
 	// Check if the current and desired referenced object are compatible.
-	if err := check.ReferencedObjectsAreStrictlyCompatible(current, desired); err != nil {
-		return err
+	if allErrs := check.ObjectsAreStrictlyCompatible(current, desired); len(allErrs) != 0 {
```

Comment:
```suggestion
	if allErrs := check.ObjectsAreStrictlyCompatible(current, desired); len(allErrs) > 0 {
```
Maybe better as it better expresses what we want to check and negative lengths are not possible (same for other cases)